---
title: "Deconstructing a complex Lens - functional implications of cataract induction in an insect visual system"
authors: "Mitra AT., Rathore S., Jester A., Hyland-Brown R., Hassert J., Benoit JB., Stowasser A., Buschbeck EK."
date: "7 April 2024"
output:
  html_document:
    
    theme: flatly
    toc: TRUE
    toc_float: TRUE
    code_fold: hide
editor_options: 
  chunk_output_type: console
---

```{r setup, include = FALSE}

# Markdown settings
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, fig.height = 6, fig.width = 8, attr.output='style="max-height: 400px;"') 

# Load packages
library(tidyverse)
library(cowplot)
library(reshape2)
library(grid)
library(gridExtra)
library(plotly)
library(ggpubr)
library(ggformula)
library(ggplotify)
library(magick)
library(rstatix)
library(Cairo)

```

# Data and subsetting

## Raw data

Raw data files for the ophthalmoscope, edge sharpness and focal length contain one row for **each eye** sampled. For most individuals, there are four rows of data: Right eye 1 & eye 2 and left eye 1 & eye 2. 

Raw data file for behaviour contains data from both the optimal and dim light/challenging light arenas.

```{r loading data}

# Load raw data

# qPCR
qpcr_data <- read.csv("lens3_qPCR.csv", header = TRUE)

# Hanging drops
edge_data <- read.csv("lens3_edge_sharpness_long.csv", header = TRUE) # raw edge sharpness values
hang_data <- read.csv("lens3_hanging_drops.csv", header = TRUE) # peaks and maximum sharpness values extracted from raw edge sharpness data ^

# Ophthalmoscope
oph_data <- read.csv("lens3_ophthalmoscope.csv", header = TRUE)

# Behaviour
behaviour_data <- read.csv("lens3_behaviour.csv", header = TRUE)

# Behaviour arena spectrophotometer
spec_data <- read.csv("behaviour_arena_spec.csv", header = TRUE)

# na.strings converts "NA", "" and " " into na within the body of the file

# Colourblind friendly palette
palette.colors(palette = "Okabe-Ito")
```

## qPCR

Comparing lens3 expression using qPCR on whole-head cDNA.

```{r qPCR plot}

# ATM generated data & STR analysed

# Boxplot of log fold gene expression
qPCR_plot <- qpcr_data %>%
  ggplot(aes(y= log_fold_gene_expression, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  labs(title=NULL,x = NULL, y = "Log fold gene expression") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  geom_signif(comparisons=list(c("control", "lens3_249")), annotations="*", # adds a significance star
              map_signif_level = TRUE, textsize = 10, 
              vjust = 0.5,
              margin_top = 0.1) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20),
                                 legend.position = "none")
qPCR_plot
```

# Figure 2
```{r figure 2: qPCR }

pdf(file = "Figures/figure_2.pdf", width = 9,  height = 6)
qPCR_plot
dev.off()

```

## Relative Edge Sharpness 

Comparing the relative edge sharpness curves across groups using LOESS, which is a locally weighted scatterplot smoother. The method fits a smooth line through data by fitting weighted least squares (WLS) models to observations within a user-specified window of the focal point, whose width (span) is typically expressed as a proportion α of the n data points. The ribbon is representative of standard error.  

```{r relative edge sharpness curves}

# Creating new column with the eye and probe identity
edge_data <- edge_data %>%
  mutate(eye_probe = paste(eye, probe, sep="_"))

# Trimming dataset to 550 µm for consistency
edge_data <- edge_data %>%
  filter(distance <= 600)

# Individual eye plots

# LE1
LE1_edge_plot <- edge_data %>%
  filter(eye == "LE1") %>%
  ggplot(aes(y= contrast, x = distance, group = eye_probe)) +
  #geom_point() +
  stat_smooth(span = 0.1, se = TRUE,
              method = 'loess',
              formula = 'y ~ x',
              aes(fill = eye_probe, colour = eye_probe)) +
   scale_colour_manual(values = c("#009E73", "#E69F00", "#D55E00"), 
                    labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_fill_manual(values = c("#009E73", "#E69F00", "#D55E00"), 
                    labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) + # For some reason, if labels are not specified in both colour and fill, they are plotted twice. Weird
  scale_x_continuous("Distance from back surface of lens (µm)", 
                     breaks = c(250, 350, 450, 550), 
                     limits = c(185, 600)) +
  scale_y_continuous("Relative edge sharpness",
                     breaks = c(1, 1.5, 2.5, 3.5)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -1.2),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20),
        legend.title = element_blank(),
        legend.text = element_text(size = 15))
LE1_edge_plot 

# LE2
LE2_edge_plot <- edge_data %>%
  filter(eye == "LE2") %>%
  ggplot(aes(y= contrast, x = distance, group = eye_probe)) +
  #geom_point() +
  stat_smooth(span = 0.1, se = TRUE,
              method = 'loess',
              formula = 'y ~ x',
              aes(fill = eye_probe, colour = eye_probe)) +
  scale_colour_manual(values = c("#009E73", "#E69F00", "#D55E00"), 
                    labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_fill_manual(values = c("#009E73", "#E69F00", "#D55E00"), 
                    labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) + # For some reason, if labels are not specified in both colour and fill, they are plotted twice. Weird
  scale_x_continuous("Distance from back surface of lens (µm)", 
                     breaks = c(250, 350, 450, 550), 
                     limits = c(185, 600)) +
  scale_y_continuous("Relative edge sharpness",
                     breaks = c(1.5, 2.5, 3.5, 4.5)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -1.2),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 25),
        legend.title = element_blank(),
        legend.text = element_text(size = 18),
        legend.position = "inside", legend.position.inside = c(0.9, 0.9), # setting legend coordinates to inside the plot
        legend.key.size = unit(0.7, 'cm'), #change legend key size
              legend.key.height = unit(0.7, 'cm'), #change legend key height
              legend.key.width = unit(0.7, 'cm')) #change legend key width
      
LE2_edge_plot 

# RE1
RE1_edge_plot <- edge_data %>%
  filter(eye == "RE1") %>%
  ggplot(aes(y= contrast, x = distance, group = eye_probe)) +
  #geom_point() +
  stat_smooth(span = 0.1, se = TRUE, 
              method = 'loess',
              formula = 'y ~ x',
              aes(fill = eye_probe, colour = eye_probe)) +
  scale_colour_manual(values = c("#009E73", "#E69F00", "#D55E00"), 
                    labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_fill_manual(values = c("#009E73", "#E69F00", "#D55E00"), 
                    labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) + # For some reason, if labels are not specified in both colour and fill, they are plotted twice. Weird
  scale_x_continuous("Distance from back surface of lens (µm)", 
                     breaks = c(250, 350, 450, 550), 
                     limits = c(185, 600)) +
  scale_y_continuous("Relative edge sharpness",
                     breaks = c(1, 1.5, 2.5, 3.5)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -1.2),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20),
        legend.title = element_blank(),
        legend.text = element_text(size = 15))
RE1_edge_plot 

# RE2
RE2_edge_plot <- edge_data %>%
  filter(eye == "RE2") %>%
  ggplot(aes(y= contrast, x = distance, group = eye_probe)) +
  #geom_point() +
  stat_smooth(span = 0.1, se = TRUE, 
              method = 'loess',
              formula = 'y ~ x',
              aes(fill = eye_probe, colour = eye_probe)) +
  scale_colour_manual(values = c("#009E73", "#E69F00", "#D55E00"), 
                    labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_fill_manual(values = c("#009E73", "#E69F00", "#D55E00"), 
                    labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) + # For some reason, if labels are not specified in both colour and fill, they are plotted twice. Weird
  scale_x_continuous("Distance from back surface of lens (µm)", 
                     breaks = c(250, 350, 450, 550), 
                     limits = c(185, 600)) +
  scale_y_continuous("Relative edge sharpness",
                     breaks = c(1.5, 2.5, 3.5, 4.5)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -1.2),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20),
        legend.title = element_blank(),
        legend.text = element_text(size = 15))
RE2_edge_plot 

# Plot for a single individual
individual_edge_plot <- edge_data %>%
  filter(individual == "C12") %>%
  ggplot(aes(y= contrast, x = distance, group = eye)) +
  geom_point(aes(fill = eye_probe, colour = eye_probe)) +
  scale_colour_manual(values = c("#56B4E9", "#E69F00", "#0072B2", "#D55E00")) +
  scale_fill_manual(values = c("#56B4E9", "#E69F00", "#0072B2", "#D55E00")) + # For some reason, if labels are not specified in both colour and fill, they are plotted twice. Weird
  scale_x_continuous("Distance from back surface of lens (µm)") +
  scale_y_continuous("Relative edge sharpness") +
  theme_classic() +
  theme(axis.title.x = element_text(size = 15, face = "bold", vjust = -1.2),
        axis.title.y = element_text(size = 15, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 15, face = "bold"),
        legend.title = element_blank(),
        legend.text = element_text(size = 15))
ggplotly(individual_edge_plot) 

```

Comparing maximum edge sharpness values between controls and tests. This usually corresponds to the second peak/image. However, sometimes the first peak has higher edge sharpness. This happens more often in knockdowns; That's some of the eyes of some individuals (3/10) for Probe 1 and for Probe 2 (4/10) and just right eyes in one control individual. In some eyes of Probe 2 individuals (2/10; 249.13 - LE2, 249.17 - LE1, RE1), best focused images could not be detected in the edge sharpness plots or manually

```{r maximum edge sharpness boxplots}

# LE1
LE1_max_edge <- hang_data %>%
  filter(eye == "LE1") %>%
  ggplot(aes(y= maximum_sharpness, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_y_continuous(limits = c(0, 8.2), breaks = c(1, 3, 5, 7)) +
  labs(title=NULL,x="Left eye 1", y = "Maximum edge sharpness") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  geom_signif(comparisons=list(c("control", "lens3_249")), annotations="*", # adds a significance star
              map_signif_level = TRUE, textsize = 10, 
              vjust = 0.5,
              margin_top = 0.1) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")
LE1_max_edge

# LE2
LE2_max_edge <- hang_data %>%
  filter(eye == "LE2") %>%
  ggplot(aes(y= maximum_sharpness, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_y_continuous(limits = c(0, 8.75), breaks = c(2, 4, 6, 8)) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  labs(title=NULL,x="Left eye 2", y = "Maximum edge sharpness") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) +
  geom_signif(comparisons=list(c("control", "lens3_249")), annotations="*", # adds a significance star
              map_signif_level = TRUE, textsize = 10, 
              vjust = 0.5,
              margin_top = 0.1) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 19),
                                 legend.position = "none")
LE2_max_edge

# RE1
RE1_max_edge <- hang_data %>%
  filter(eye == "RE1") %>%
  ggplot(aes(y= maximum_sharpness, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_y_continuous(limits = c(0, 8.2), breaks = c(1, 3, 5, 7)) +
  labs(title=NULL,x="Right eye 1", y = "Maximum edge sharpness") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) +
  geom_signif(comparisons=list(c("control", "lens3_249")), annotations="*", # adds a significance star
              map_signif_level = TRUE, textsize = 10, 
              vjust = 0.5,
              margin_top = 0.1) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")
RE1_max_edge

# RE2
RE2_max_edge <- hang_data %>%
  filter(eye == "RE2") %>%
  ggplot(aes(y= maximum_sharpness, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_y_continuous(limits = c(0, 8.75), breaks = c(2, 4, 6, 8)) +
  labs(title=NULL,x="Right eye 2", y = "Maximum edge sharpness") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  geom_signif(comparisons=list(c("control", "lens3_249")), annotations="*", # adds a significance star
              map_signif_level = TRUE, textsize = 10, 
              vjust = 0.5,
              margin_top = 0.1) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")
RE2_max_edge

```

## Back focal length
Using the 2nd (typically taller) peak. Verified through visual assessment of the image series.

```{r back focal length boxplots}

# LE1
LE1_focal <- hang_data %>%
  filter(eye == "LE1") %>%
  ggplot(aes(y= peak_2_um, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_y_continuous(limits = c(250, 500), 
                     breaks = c(300, 350, 400, 450, 500)) +
  labs(title=NULL,x="Left eye 1", y = "Back focal length (µm)") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")
LE1_focal

# LE2
LE2_focal <- hang_data %>%
  filter(eye == "LE2") %>%
  ggplot(aes(y= peak_2_um, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_y_continuous(limits = c(250, 500), 
                     breaks = c(300, 350, 400, 450, 500)) +
  labs(title=NULL,x="Left eye 2", y = "Back focal length (µm)") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 30, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 30, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 19),
                                 legend.position = "none")
LE2_focal

# RE1
RE1_focal <- hang_data %>%
  filter(eye == "RE1") %>%
  ggplot(aes(y= peak_2_um, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_y_continuous(limits = c(250, 500), 
                     breaks = c(300, 350, 400, 450, 500)) +
  labs(title=NULL,x="Right eye 1", y = "Back focal length") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")
RE1_focal

# RE2
RE2_focal <- hang_data %>%
  filter(eye == "RE2") %>%
  ggplot(aes(y= peak_2_um, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_y_continuous(limits = c(250, 500), 
                     breaks = c(300, 350, 400, 450, 500)) +
  labs(title=NULL,x="Right eye 2", y = "Back focal length") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")
RE2_focal

```

# Figure 3
```{r figure 3: sharpness curve & maximum sharpness & focal length}

# making individual panels 
fig.f <- plot_grid(LE2_edge_plot,
                   labels = "F",
                   label_size = 29.5,
                   hjust = 1.2, # adjustment for panel labels
                   vjust = 0.1) +
  theme(plot.margin = unit(c(20, 25, 20, 20), "points")) # The order of edges for plot.margin is unit(c(top, right, bottom, left), units)
fig.f
  
fig.b <- LE2_max_edge +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.title.y = element_text(size = 25)) +
  theme(plot.margin = unit(c(20, 20, 20, 20), "points"))
fig.b

fig.c <- LE2_focal +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_text(size = 25)) +
  theme(plot.margin = unit(c(20, 20, 20, 20), "points"))
fig.c

# stacking panel g and h in one column
fig.gh <- plot_grid(fig.b, NULL, fig.c,
                     rel_heights = c(6, 0.1, 6),
                     align = "v",
                     ncol = 1,
                     labels = c("G", "", "H"),
                     label_size = 29.5,
                     hjust = 0.8, # adjustment for panel labels
                     vjust = 0.1) +
  theme(plot.margin = unit(c(20, 20, 5, 20), "points")) # The order of edges for plot.margin is unit(c(top, right, bottom, left), units)

fig.gh

# assembling the final figure
fig.fgh <- plot_grid(fig.f, fig.gh,
                     rel_widths = c(1.2, 1),
                     nrow = 1)
fig.fgh

# writing to pdf
pdf(file = "Figures/figure_3.pdf", width = 20,  height = 10)
fig.fgh
dev.off()

```

# Figure S1
```{r figure S1: hanging drops}

###### edge sharpness curves - row 1 ######
fig.a <- LE1_edge_plot +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank()) + 
  theme(legend.position = "none",
        plot.margin = unit(c(0.5, 0, 0.9, 0.7), "cm")) #The order of edges for plot.margin is unit(c(top, right, bottom, left), units)

# adding a second x axis label
fig.a <- ggdraw(fig.a) + 
  draw_label("Left eye 1",
             size = 20,
             y = 0.05, # y coordinates
             x = 0.55) 
  
fig.a

fig.b <- RE1_edge_plot +
 theme(axis.title.x = element_blank(),
        axis.title.y = element_blank()) + 
  theme(legend.position = "none",
        plot.margin = unit(c(0.5, 0, 0.9, 0.7), "cm"))

# adding a second x axis label
fig.b <- ggdraw(fig.b) + 
  draw_label("Right eye 1",
             size = 20,
             y = 0.05, # y coordinates
             x = 0.55) 
  
fig.b

fig.c <- RE2_edge_plot +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank()) + 
  theme(legend.position = "none",
        plot.margin = unit(c(0.5, 0, 0.9, 0.7), "cm"))
  
# adding a second x axis label
fig.c <- ggdraw(fig.c) + 
  draw_label("Right eye 2",
             size = 20,
             y = 0.05, # y coordinates
             x = 0.55) 

fig.c

# Creating common axis labels
y.grob <- textGrob("Relative edge sharpness", 
                   gp = gpar(fontface = "bold",
                             fontsize = 25), rot = 90,
                             vjust = 0.8,
                             hjust = 0.4)

x.grob <- textGrob("Distance from back surface of lens (µm)",
                   gp = gpar(fontface = "bold",
                             fontsize = 25))

# extract legend from one figure
legend <- get_legend(LE1_edge_plot + 
                       theme(legend.position = "bottom")) # legend is laid out horizontally

# Use cowplot package to nicely plot the graphs together
edge_supplement_plot_trim <- plot_grid(fig.a, fig.b, fig.c,
                   #list of plots to arrange in grid
                   labels = c("A","B","C"), # panel labels for figure
                   label_size = 23,
                   hjust = 0.3, # adjustment for panel labels
                   vjust = 0.9,
                   align = "v",
                   nrow = 1) +
  theme(plot.margin = unit(c(20, 20, 5, 20), "points")) # The order of edges for plot.margin is unit(c(top, right, bottom, left), units)

plot_grid(edge_supplement_plot_trim)

edge_supplement_xlab_trim <- plot_grid(edge_supplement_plot_trim, x.grob,
                                 ncol = 1, rel_heights = c(1, .1))
plot_grid(edge_supplement_xlab_trim)

edge_supplement_final <- grid.arrange(arrangeGrob(edge_supplement_xlab_trim, 
                         left = y.grob,
                         bottom = legend))

###### maximum edge sharpness boxplots - row 2 ######
fig.d <- LE1_max_edge +
  theme(axis.title.y = element_blank(),
        axis.title.x = element_blank()) + 
  theme(legend.position = "none",
        plot.margin = unit(c(0.5, 0, 0.5, 0.7), "cm")) #The order of edges for plot.margin is unit(c(top, right, bottom, left), units)
  
fig.d

fig.e <- RE1_max_edge +
  theme(axis.title.y = element_blank(),
        axis.title.x = element_blank()) + 
  theme(legend.position = "none",
        plot.margin = unit(c(0.5, 0, 0.5, 0.7), "cm"))
  
fig.e

fig.f <- RE2_max_edge +
  theme(axis.title.y = element_blank(),
        axis.title.x = element_blank()) + 
  theme(legend.position = "none",
        plot.margin = unit(c(0.5, 0, 0.5, 0.7), "cm"))
  
fig.f

# Creating common axis labels
y.grob <- textGrob("Maximum edge sharpness", 
                   gp = gpar(fontface = "bold",
                             fontsize = 25), rot = 90,
                             vjust = 0.6,
                             hjust = 0.5)

# Use cowplot package to nicely plot the graphs together 
max_sharpness_plot_trim <- plot_grid(fig.d, fig.e, fig.f,
                   #list of plots to arrange in grid
                   align = "h",
                   labels = c("D","E","F"), # panel labels for figure
                   label_size = 23,
                   hjust = 0.3, # adjustment for panel labels
                   vjust = 0.9,
                   nrow = 1) +
  theme(plot.margin = unit(c(20, 20, 5, 20), "points")) # The order of edges for plot.margin is unit(c(top, right, bottom, left), units)

plot_grid(max_sharpness_plot_trim)

max_sharpness_final <- grid.arrange(arrangeGrob(max_sharpness_plot_trim, left = y.grob))

####### back focal length boxplots - row 3
fig.g <- LE1_focal +
  theme(axis.title.y = element_blank()) + 
  theme(legend.position = "none",
        plot.margin = unit(c(0.5, 0, 0.5, 0.7), "cm")) # The order of edges for plot.margin is unit(c(top, right, bottom, left), units)
  
fig.g

fig.h <- RE1_focal +
  theme(axis.title.y = element_blank()) + 
  theme(legend.position = "none",
        plot.margin = unit(c(0.5, 0, 0.5, 0.7), "cm")) #The order of edges for plot.margin is unit(c(top, right, bottom, left), units)
  
fig.h

fig.i <- RE2_focal +
  theme(axis.title.y = element_blank()) + 
   theme(legend.position = "none",
        plot.margin = unit(c(0.5, 0, 0.5, 0.7), "cm"))
  
fig.i

# Creating common axis labels
y.grob <- textGrob("Back focal length (µm)", 
                   gp = gpar(fontface = "bold",
                             fontsize = 25), rot = 90, 
                             vjust = 0.6,
                             hjust = 0.5)

# Use cowplot package to nicely plot the graphs together 
focal_supplement_plot_trim <- plot_grid(fig.g, fig.h, fig.i,
                   #list of plots to arrange in grid
                   rel_widths = c(6, 6),
                   labels = c("G","H","I"), # panel labels for figure
                   label_size = 23,
                   hjust = 0.3, # adjustment for panel labels
                   vjust = 0.9,
                   nrow = 1) +
  theme(plot.margin = unit(c(20, 20, 5, 20), "points")) # The order of edges for plot.margin is unit(c(top, right, bottom, left), units)

plot_grid(focal_supplement_plot_trim)

focal_supplement_final <- grid.arrange(arrangeGrob(focal_supplement_plot_trim, left = y.grob))

###### assembling final figure ######
final.supplement <- plot_grid(edge_supplement_final,
                              max_sharpness_final,
                              focal_supplement_final,
                     rel_heights = c(1.2, 1, 1),
                     nrow = 3)
final.supplement

# writing to pdf
pdf(file = "Figures/figure_S1.pdf", width = 19,  height = 14)
final.supplement
dev.off()

```

## Ophthalmoscope

```{r individual eye boxplots}

# LE1
LE1_oph <- oph_data %>%
  filter(eye == "LE1") %>%
  ggplot(aes(y= bestimagereciprocal, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  labs(title=NULL,x="Left Eye 1", y = "Object distance (mm)") +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_y_continuous(breaks = c(-0.2, -0.1, 0, 0.1, 0.2),
                     label = c("5", "10", "∞", "10","5")) + # reciprocal for image distance
  geom_jitter(shape = 19, size = 3, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 30, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 30, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20),
                                 legend.position = "none")
LE1_oph

# LE2
LE2_oph <- oph_data %>%
  filter(eye == "LE2") %>%
  ggplot(aes(y= bestimagereciprocal, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  labs(title=NULL,x="Left Eye 2", y = "Object distance (mm)") +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_y_continuous(breaks = c(-0.2, -0.1, 0, 0.1, 0.2),
                     label = c("5", "10", "∞", "10","5")) +
  geom_jitter(shape = 19, size = 3, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 30, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 30, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20),
                                 legend.position = "none")
LE2_oph

# RE1
RE1_oph <- oph_data %>%
  filter(eye == "RE1") %>%
  ggplot(aes(y= bestimagereciprocal, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  labs(title=NULL,x="Right Eye 1", y = "Object distance (mm)") +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_y_continuous(breaks = c(-0.2, -0.1, 0, 0.1, 0.2),
                     label = c("5", "10", "∞", "10","5")) +
  geom_jitter(shape = 19, size = 3, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 30, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 30, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20),
                                 legend.position = "none")
RE1_oph

# RE2

RE2_oph <- oph_data %>%
  filter(eye == "RE2") %>%
  ggplot(aes(y= bestimagereciprocal, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = 17, outlier.size = 2, position = "dodge2") + # Changing outlier shape
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  labs(title=NULL,x="Right Eye 2", y = "Object distance (mm)") +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi P1"), 
                                         paste(italic("lens3"), "RNAi P2")))) +
  scale_y_continuous(breaks = c(-0.2, -0.1, 0, 0.1, 0.2),
                     label = c("5", "10", "∞", "10","5")) +
  geom_jitter(shape = 19, size = 3, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 30, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 30, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20),
                                 legend.position = "none")
RE2_oph

# Making the individual boxes narrow and putting them next to each other like the osmosis paper figures

# Creating new column with the eye and probe identity to make this easier. Might be a better way to do this?
oph_data <- oph_data %>%
  mutate(eye_probe = paste(eye, probe, sep="_"))

# Reordering factors so control and probe boxes are grouped together
oph_data$eye_probe <- factor(oph_data$eye_probe,
                          c("LE1_control", "RE1_control", "LE2_control", "RE2_control", "LE1_lens3_191", "RE1_lens3_191", "LE2_lens3_191", "RE2_lens3_191", "LE1_lens3_249", "RE1_lens3_249", "LE2_lens3_249", "RE2_lens3_249"))

# Plot for all E1s

E1_LR_oph <- oph_data %>%
  filter(grepl('E1', eye)) %>% #grepl stands for grep logical. It is a function that searches for matches or a string or a string vector
  ggplot(aes(y= bestimagereciprocal, x = eye_probe, fill = eye_probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#009E73", "#E69F00",  "#E69F00", "#D55E00", "#D55E00")) +
  labs(title=NULL,x="Eye 1", y = "Object distance (mm)") +
  scale_x_discrete(labels = c('Left', 'Right', 'Left', 'Right', 'Left', 'Right')) +
  scale_y_continuous(breaks = c(-0.2, -0.1, 0, 0.1, 0.2),
                     label = c("5", "10", "∞", "10","5")) +
  geom_jitter(shape = 19, size = 3, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 30, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 30, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20),
                                 legend.position = "none")
E1_LR_oph 

# Making custom facet grid labels for use with the labeller function
probe.labs <- c("control" = "Control", "lens3_191" = "lens3RNAi P1", "lens3_249" = "lens3RNAi P2")

# Making multilevel x-axes

E1_LR_oph <- E1_LR_oph + facet_grid(~ probe, scales = "free", labeller = labeller(probe = probe.labs)) 

# Changing text size of facet grid labels
E1_LR_oph <- E1_LR_oph + theme(strip.text.x = element_text(size = 20, face = "bold"))

# Customising facet appearance
E1_LR_oph <- E1_LR_oph + theme(strip.background = element_rect(size = 1.5, linetype = "solid"))

E1_LR_oph

# Adding colours to facet labels

# First removing the y axis label because I cannot for the life of me figure out why this isn't straightforward to do with theme() after converting the grob back to a ggplot object.

E1_LR_oph_fig <- E1_LR_oph + theme(axis.title.y = element_blank())

E1_g <- ggplot_gtable(ggplot_build(E1_LR_oph_fig)) # converting the ggplot list to a grob and modifying it
strip_t <- which(grepl('strip-', E1_g$layout$name))
fills <- c("#009E73","#E69F00","#D55E00")
k <- 1
for (i in strip_t) {
j <- which(grepl('rect', E1_g$grobs[[i]]$grobs[[1]]$childrenOrder))
E1_g$grobs[[i]]$grobs[[1]]$children[[j]]$gp$fill <- fills[k]
k <- k+1
}
grid.draw(E1_g)

# Converting the grob to a ggplot object to align plots with cowplot
E1_LR_oph_fig <- as.ggplot(E1_g) 

# Plot for all E2s

E2_LR_oph <- oph_data %>%
  filter(grepl('E2', eye)) %>% #grepl stands for grep logical. It is a funciton that searches for matches or a string or a string vector
  ggplot(aes(y= bestimagereciprocal, x = eye_probe, fill = eye_probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#009E73", "#E69F00", "#E69F00", "#D55E00", "#D55E00")) +
  labs(title=NULL,x="Eye 2", y = "Object distance (mm)") +
  scale_x_discrete(labels = c('Left', 'Right', 'Left', 'Right', 'Left', 'Right')) +
  scale_y_continuous(breaks = c(-0.2, -0.1, 0, 0.1, 0.2),
                     label = c("5", "10", "∞", "10","5")) +
  geom_jitter(shape = 19, size = 3, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 30, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 30, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20),
                                 legend.position = "none")
E2_LR_oph 

# Trying to make multilevel x-axes
E2_LR_oph <- E2_LR_oph + facet_grid(~ probe, scales = "free", labeller = labeller(probe = probe.labs)) 

# Changing text size of facet grid labels
E2_LR_oph <- E2_LR_oph + theme(strip.text.x = element_text(size = 20, face = "bold"))

# Customising facet appearance
E2_LR_oph <- E2_LR_oph + theme(strip.background = element_rect(size = 1.5, linetype = "solid"))

E2_LR_oph

```

# Figure 6
```{r figure 6: ophthalmoscope}

# Make panels 
fig.a <- E1_LR_oph +
theme(legend.position = "none")

fig.a

fig.b <- E2_LR_oph +
theme(legend.position = "none") +
  labs(y = NULL) + #removes y axis label
  guides(y = "none") #removes y axis line and numbers

fig.b

# Use cowplot package to nicely plot the graphs together
wide_plot <- plot_grid(fig.a, NULL, fig.b, 
                   #list of plots to arrange in grid
                   rel_widths = c(6, 0.2, 6),
                   nrow = 1)

plot_grid(wide_plot)

# Stacking the plots on top of each other instead of putting them side to side
fig.a <- E1_LR_oph_fig +
theme(legend.position = "none",
      axis.title.y = element_blank(),
      axis.text.y = element_blank()) +
  labs(y = NULL) +
  theme(plot.margin = unit(c(20, 10, 10, 20), "points")) # The order of edges for plot.margin is unit(c(top, right, bottom, left), units)

fig.a

fig.b <- E2_LR_oph +
theme(legend.position = "none",
      strip.text.x = element_blank())+ #removing facet_grid second x axis label
  labs(y = NULL) +
  theme(plot.margin = unit(c(10, 10, 20, 20), "points"))

fig.b

# Combining plots

long_plot <- plot_grid(fig.a, fig.b, 
                       align = 'v', 
                       axis = 'r',
                       nrow = 2,
                       rel_heights = c(7, 6))
plot_grid(long_plot)

# Creating common y axis label
y.grob <- textGrob("Object distance (mm)", 
                   gp = gpar(fontface = "bold",
                             fontsize = 30), rot = 90)

# Print graph
cairo_pdf(file = "Figures/figure_6.pdf", width = 12,  height = 9) # cairo_pdf is able to handle special characters
grid.arrange(arrangeGrob(long_plot, left = y.grob)) #adding common y axis label to plot
dev.off()
```

```{r estimating left-right asymmetry}

# making a dataframe with one row per individual

# subset each eye
LE1_oph_data <- filter(oph_data, eye == "LE1")
RE1_oph_data <- filter(oph_data, eye == "RE1")
LE2_oph_data <- filter(oph_data, eye == "LE2")
RE2_oph_data <- filter(oph_data, eye == "RE2")

# putting all dataframes into a list
oph_df_list <- list(LE1_oph_data, RE1_oph_data, LE2_oph_data, RE2_oph_data)

# merging all dataframes in list 
LR_oph_data <- oph_df_list %>% 
  reduce(full_join, by = 'individual')

# clean up
LR_oph_data <- LR_oph_data %>%
  rename(bestimagereciprocal.LE1 = bestimagereciprocal.x) %>% # renaming columns
  rename(bestimagereciprocal.RE1 = bestimagereciprocal.y) %>%
  rename(bestimagereciprocal.LE2 = bestimagereciprocal.x.x) %>%
  rename(bestimagereciprocal.RE2 = bestimagereciprocal.y.y) %>%
  rename(group = group.x) %>%
  rename(probe = probe.x) %>%
  select( -c(eye.x, eye.x.x, eye.y, eye.y.y, # removing redundant columns
             group.x.x, group.y, group.y.y,
             probe.x.x, probe.y, probe.y.y))

# not by any means an elegant way to do this but I couldn't figure anything else out so :P

# linear models of left vs right reciprocal object distance for each probe
## E1
LR_E1_control_mod <- lm(bestimagereciprocal.LE1 ~ bestimagereciprocal.RE1, 
                        data = LR_oph_data,
                        subset = probe == "control")

LR_E1_191_mod <- lm(bestimagereciprocal.LE1 ~ bestimagereciprocal.RE1, 
                        data = LR_oph_data,
                        subset = probe == "lens3_191")
LR_E1_249_mod <- lm(bestimagereciprocal.LE1 ~ bestimagereciprocal.RE1, 
                        data = LR_oph_data,
                        subset = probe == "lens3_249")

# linear models of left vs right reciprocal object distance for each probe
## E2
LR_E2_control_mod <- lm(bestimagereciprocal.LE2 ~ bestimagereciprocal.RE2, 
                        data = LR_oph_data,
                        subset = probe == "control")

LR_E2_191_mod <- lm(bestimagereciprocal.LE2 ~ bestimagereciprocal.RE2, 
                        data = LR_oph_data,
                        subset = probe == "lens3_191")
LR_E2_249_mod <- lm(bestimagereciprocal.LE2 ~ bestimagereciprocal.RE2, 
                        data = LR_oph_data,
                        subset = probe == "lens3_249")

# summary of each model
## E1
summary(LR_E1_control_mod)
# adj r.r = 0.015
summary(LR_E1_191_mod)
# adj r.r = 0.008
summary(LR_E1_249_mod)
# adj r.r = -0.124

## E2
summary(LR_E2_control_mod)
# adj r.r = 0.80 ***
summary(LR_E2_191_mod)
# adj r.r = 0.304
summary(LR_E2_249_mod)
# adj r.r = -0.101
```

```{r left-right asymmetry plots}

# left eye vs right eye for object distance 

# E1
## control
E1_LR_control_oph_plot <- LR_oph_data %>%
  filter(probe == "control") %>%
  ggplot(aes(y= bestimagereciprocal.LE1, x = bestimagereciprocal.RE1)) +
  labs(title=NULL, x = "Right eye 1", y = "Left eye 1") +
  geom_smooth(method = "lm", color = "#009E73", fill = "#009E73") +
  geom_point(alpha = 0.7, size = 2.5) +
  scale_x_continuous(breaks = c(-0.2, -0.1, 0, 0.1, 0.2, 0.3),
                     label = c("5", "10", "∞", "10","5", "3.3")) +
  scale_y_continuous(breaks = c(-0.2, 0, 0.2, 0.4),
                     label = c("5", "∞", "5", "2.5")) +
  theme_classic() +
  annotate(geom="text", x = 0.25, y = 0.5, 
           label= expression("R"^2*" = 0.015"),
           size = 7) +
  theme(axis.title.x = element_text(size = 23, vjust = -0.15),
        axis.title.y = element_text(size = 23, vjust = 1.8),
        axis.text = element_text(size = 20))

E1_LR_control_oph_plot

## Probe 1
E1_LR_191_oph_plot <- LR_oph_data %>%
  filter(probe == "lens3_191") %>%
  ggplot(aes(y= bestimagereciprocal.LE1, x = bestimagereciprocal.RE1)) +
  labs(title=NULL, x="Right eye 1", y = "Left eye 1") +
  geom_smooth(method = "lm", color = "#E69F00", fill = "#E69F00") +
  geom_point(alpha = 0.7, size = 2.5) +
  scale_x_continuous(breaks = c(-0.2, -0.1, 0, 0.1, 0.2),
                     label = c("5", "10", "∞", "10","5")) +
  scale_y_continuous(breaks = c(-0.2, 0, 0.2, 0.4),
                     label = c("5", "∞", "5", "2.5")) +
  theme_classic() +
  annotate(geom="text", x = 0.125, y = 0.6, 
           label= expression("R"^2*" = 0.008"),
           size = 7) +
  theme(axis.title.x = element_text(size = 23, vjust = -0.15),
        axis.title.y = element_text(size = 23, vjust = 1.8),
        axis.text = element_text(size = 20))

E1_LR_191_oph_plot

## Probe 2
E1_LR_249_oph_plot <- LR_oph_data %>%
  filter(probe == "lens3_249") %>%
  ggplot(aes(y= bestimagereciprocal.LE1, x = bestimagereciprocal.RE1)) +
  labs(title=NULL, x="Right eye 1", y = "Left eye 1") +
  geom_smooth(method = "lm", color = "#D55E00", fill = "#D55E00") +
  geom_point(alpha = 0.7, size = 2.5) +
  scale_x_continuous(breaks = c(-0.2, -0.1, 0, 0.1, 0.2),
                     label = c("5", "10", "∞", "10","5")) +
  scale_y_continuous(breaks = c(-0.1, 0, 0.1, 0.2),
                     label = c("10", "∞", "10", "5")) +
  theme_classic() +
  annotate(geom="text", x = 0.185, y = 0.25, 
           label= expression("R"^2*" = -0.124"),
           size = 7) +
  theme(axis.title.x = element_text(size = 23, vjust = -0.15),
        axis.title.y = element_text(size = 23, vjust = 1.8),
        axis.text = element_text(size = 20))

E1_LR_249_oph_plot

# E2
## control
E2_LR_control_oph_plot <- LR_oph_data %>%
  filter(probe == "control") %>%
  ggplot(aes(y= bestimagereciprocal.LE2, x = bestimagereciprocal.RE2)) +
  labs(title=NULL, x="Right eye 2", y = "Left eye 2") +
  geom_smooth(method = "lm", color = "#009E73", fill = "#009E73") +
  geom_point(alpha = 0.7, size = 2.5) +
  scale_x_continuous(breaks = c(-0.1, 0, 0.1, 0.2),
                     label = c("10", "∞", "10","5")) +
  scale_y_continuous(breaks = c(-0.2, 0, 0.2, 0.4),
                     label = c("5", "∞", "5", "2.5")) +
  theme_classic() +
  annotate(geom="text", x = 0.22, y = 0.41, 
           label= expression(bold("R"^2*" = 0.8")),
           size = 7) +
  theme(axis.title.x = element_text(size = 23, vjust = -0.15),
        axis.title.y = element_text(size = 23, vjust = 1.8),
        axis.text = element_text(size = 20))

E2_LR_control_oph_plot

## Probe 1
E2_LR_191_oph_plot <- LR_oph_data %>%
  filter(probe == "lens3_191") %>%
  ggplot(aes(y= bestimagereciprocal.LE2, x = bestimagereciprocal.RE2)) +
  labs(title=NULL, x="Right eye 2", y = "Left eye 2") +
  geom_smooth(method = "lm", color = "#E69F00", fill = "#E69F00") +
  geom_point(alpha = 0.7, size = 2.5) +
  scale_x_continuous(breaks = c(-0.1, 0, 0.1, 0.2),
                     label = c("10", "∞", "10","5")) +
  scale_y_continuous(breaks = c(-0.2, 0, 0.2, 0.4),
                     label = c("5", "∞", "5", "2.5")) +
  theme_classic() +
  annotate(geom="text", x = 0.18, y = 0.4, 
           label= expression("R"^2*" = 0.304"),
           size = 7) +
  theme(axis.title.x = element_text(size = 23, vjust = -0.15),
        axis.title.y = element_text(size = 23, vjust = 1.8),
        axis.text = element_text(size = 20))

E2_LR_191_oph_plot

## Probe 2
E2_LR_249_oph_plot <- LR_oph_data %>%
  filter(probe == "lens3_249") %>%
  ggplot(aes(y= bestimagereciprocal.LE2, x = bestimagereciprocal.RE2)) +
  labs(title=NULL, x="Right eye 2", y = "Left eye 2") +
  geom_smooth(method = "lm", color = "#D55E00", fill = "#D55E00") +
  geom_point(alpha = 0.7, size = 2.5) +
  scale_x_continuous(breaks = c(-0.1, -0.05, 0, 0.05),
                     label = c("10", "20", "∞", "20")) +
  scale_y_continuous(breaks = c(-0.2, -0.1, 0, 0.1),
                     label = c("5", "10", "∞", "10")) +
  theme_classic() +
  annotate(geom="text", x = 0.038, y = 0.135, 
           label= expression("R"^2*" = -0.101"),
           size = 7) +
  theme(axis.title.x = element_text(size = 25, vjust = -0.15),
        axis.title.y = element_text(size = 25, vjust = 1.8),
        axis.text = element_text(size = 20))

E2_LR_249_oph_plot

```

# Figure S2
```{r figure S2: ophthalmoscope left right asymmetry}

# Making panels
fig.a <- E1_LR_control_oph_plot

fig.b <- E1_LR_191_oph_plot 

fig.c <- E1_LR_249_oph_plot 

fig.d <- E2_LR_control_oph_plot

fig.e <- E2_LR_191_oph_plot

fig.f <- E2_LR_249_oph_plot

# Creating common axis labels
y.grob <- textGrob("Object distance (mm)", 
                   gp = gpar(fontface = "bold",
                             fontsize = 25), rot = 90, 
                             vjust = 0.8,
                             hjust = 0.2)

x.grob <- textGrob("Object distance (mm)",
                   gp = gpar(fontface = "bold",
                             fontsize = 25),
                             vjust = -0.9)

# Building the figure
LR_oph_asymmetry <- plot_grid(fig.a, fig.b, fig.c, fig.d, fig.e, fig.f,
           align = 'vh', # align horizontally and vertically
           axis = 'lb',
           labels = c("A","B","C","D", "E", "F"), # panel labels for figure
           label_size = 28,
           hjust = -.1, # adjustment for panel labels
           nrow = 2) + # number of rows in grids
           theme(plot.margin = unit(c(10, 10, 20, 25), "points")) # The order of edges for plot.margin is unit(c(top, right, bottom, left), units)
  
LR_oph_asymmetry

# adding common x axis label
LR_asymmetry_xlab <- plot_grid(LR_oph_asymmetry, x.grob,
                                 ncol = 1, rel_heights = c(1, 0.04))
plot_grid(LR_asymmetry_xlab)

# extracting legend from another figure and increasing size
legend_large <- get_legend(LE1_edge_plot + 
              theme(legend.position = "bottom", # legend is laid out horizontally
              legend.key.size = unit(1.2, 'cm'), #change legend key size
              legend.key.height = unit(1.2, 'cm'), #change legend key height
              legend.key.width = unit(1.2, 'cm'), #change legend key width
              legend.text = element_text(size=18))) #change legend text font size) 

# Print graph
cairo_pdf(file = "Figures/figure_S2.pdf", width = 23,  height = 13) # cairo_pdf is able to handle special characters
grid.arrange(arrangeGrob(LR_asymmetry_xlab,
                         left = y.grob,
                         bottom = legend_large)) # legend extracted from the edge supplement curves 
dev.off()

```

## Behaviour

```{r calculating total strikes & hunting success}

# Creating total strike and percentage hunting success columns
behaviour_data <- behaviour_data %>%
  mutate(total_strikes = missed_strikes + succesful_strikes) %>%
  mutate(missed_dead_prey_strikes_NA_removed = missed_dead_prey_strikes) %>% # duplicating the missed dead prey strikes column because NAs need to be turned to 0s to properly calculate % hunting success. 
  replace_na(list(missed_dead_prey_strikes_NA_removed = 0)) %>%
  mutate(hunting_success = (succesful_strikes - (dead_prey_strikes - missed_dead_prey_strikes_NA_removed))/3 * 100)

# Creating a column with Probe 1 and Probe 2 pooled into a single lens3 category
behaviour_data <- behaviour_data %>%
  mutate(group = probe) %>% # making a duplicate probe column called group
  mutate(group = replace(group, probe == 'lens3_191' | probe == 'lens3_249', 'lens3')) # using probe as a reference, replacing 'lens3_191' and 'lens3_249 with 'lens3'
```


## Normal light conditions

### Missed strikes 
Normal light

```{r normal light missed strikes}

## horizontal arena
missed_strikes_horizontal <- behaviour_data %>%
  filter(arena == "horizontal" & light_environment == "normal") %>%
  ggplot(aes(y= missed_strikes, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "Missed Strikes") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20, face = "bold"),
                                 legend.position = "none")
missed_strikes_horizontal

## vertical arena
missed_strikes_vertical <- behaviour_data %>%
  filter(arena == "vertical" & light_environment == "normal") %>%
  ggplot(aes(y= missed_strikes, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "Missed Strikes") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20, face = "bold"),
                                 legend.position = "none")
missed_strikes_vertical
```

### Succesful strikes 
Normal light

```{r normal light succesful strikes}

## horizontal arena
succesful_strikes_horizontal <- behaviour_data %>%
  filter(arena == "horizontal" & light_environment == "normal") %>%
  ggplot(aes(y= succesful_strikes, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "Succesful Strikes") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20, face = "bold"),
                                 legend.position = "none")
missed_strikes_horizontal

## vertical arena
succesful_strikes_vertical <- behaviour_data %>%
  filter(arena == "vertical" & light_environment == "normal") %>%
  ggplot(aes(y= succesful_strikes, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "Succesful Strikes") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20, face = "bold"),
                                 legend.position = "none")
succesful_strikes_vertical


```

### Hunting success
Normal light

```{r normal light hunting success}

### separate probes

## horizontal arena
hunting_horizontal <- behaviour_data %>%
  filter(arena == "horizontal" & light_environment == "normal") %>%
  ggplot(aes(y = hunting_success, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA, position = position_dodge(0.8)) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "% Hunting success") +
  geom_dotplot(aes(fill = NULL, alpha = 0.1), binaxis = 'y', stackdir = 'center', # using geom_dotplot here instead of geom_jitter so that the points are stacked horizontally and centre aligned (stackdir). Important as there are only 3 values for hunting success and geom_jitter makes it seem like there are more.
        stackratio = 1.5, dotsize = 1.4, position = position_dodge(0.8)) + #stackratio adjusts distance between points 
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, vjust = -0.15),
        axis.title.y = element_text(size = 40, vjust = 1.8),
        axis.text = element_text(size = 30, face = "bold"),
        legend.position = "none")
hunting_horizontal

## vertical arena
hunting_vertical <- behaviour_data %>%
  filter(arena == "vertical" & light_environment == "normal") %>%
  ggplot(aes(y = hunting_success, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA, position = position_dodge(0.8)) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "% Hunting success") +
  geom_dotplot(aes(fill = NULL, alpha = 0.1), binaxis = 'y', stackdir = 'center',
        stackratio = 1.5, dotsize = 1.4, position = position_dodge(0.8)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 40, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 30, face = "bold"),
        legend.position = "none")
hunting_vertical

### pooled probes

## horizontal arena
hunting_horizontal_pooled <- behaviour_data %>%
  filter(arena == "horizontal" & light_environment == "normal") %>%
  ggplot(aes(y = hunting_success, x = group, fill = group)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA, position = position_dodge(0.8)) +
  scale_fill_manual(values=c("#009E73", "#CC79A7")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi")))) +
  labs(title=NULL,x= "", y = "% Hunting success") +
  geom_dotplot(aes(fill = NULL, alpha = 0.1), binaxis = 'y', stackdir = 'center',
        stackratio = 1.5, dotsize = 1.4, position = position_dodge(0.8)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, vjust = -0.15),
        axis.title.y = element_text(size = 38, vjust = 1.8),
        axis.text.y = element_text(size = 30),
        axis.text.x = element_text(size = 30),
        legend.position = "none")
hunting_horizontal_pooled

## vertical
hunting_vertical_pooled <- behaviour_data %>%
  filter(arena == "vertical" & light_environment == "normal") %>%
  ggplot(aes(y = hunting_success, x = group, fill = group)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA, position = position_dodge(0.8)) +
  scale_fill_manual(values=c("#009E73", "#CC79A7")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi")))) +
  labs(title=NULL,x= "", y = "% Hunting success") +
  geom_dotplot(aes(fill = NULL, alpha = 0.1), binaxis = 'y', stackdir = 'center',
        stackratio = 1.5, dotsize = 1.4, position = position_dodge(0.8)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, vjust = -0.15),
        axis.title.y = element_text(size = 38, vjust = 1.8),
        axis.text.y = element_text(size = 30),
        axis.text.x = element_text(size = 30),
        legend.position = "none")
hunting_vertical_pooled
```

### Total strikes 
Normal light

```{r normal light total strikes}

# horizontal arena
total_strikes_horizontal <- behaviour_data %>%
  filter(arena == "horizontal" & light_environment == "normal") %>%
  ggplot(aes(y= total_strikes, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "Total Strikes") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20, face = "bold"),
                                 legend.position = "none")
total_strikes_horizontal

## vertical arena
total_strikes_vertical <- behaviour_data %>%
  filter(arena == "vertical" & light_environment == "normal") %>%
  ggplot(aes(y= total_strikes, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "Total Strikes") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20, face = "bold"),
                                 legend.position = "none")
total_strikes_vertical
```

### Latency 
Normal light

```{r normal light latency}

### separate probes

## horizontal arena
latency_horizontal_separate <- behaviour_data %>%
  drop_na(latency_min) %>%
  filter(arena == "horizontal" & light_environment == "normal") %>%
  ggplot(aes(y= latency_min, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  scale_y_continuous(limits = c(0, 26), breaks = c(0, 5, 10, 15, 20, 25)) +
  labs(title=NULL,x= "", y = "Latency (min)") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 40, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 30, face = "bold"),
                                 legend.position = "none")
latency_horizontal_separate

## vertical arena
latency_vertical_separate <- behaviour_data %>%
  drop_na(latency_min) %>%
  filter(arena == "vertical" & light_environment == "normal") %>%
  ggplot(aes(y= latency_min, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  scale_y_continuous(limits = c(0, 26), breaks = c(0, 5, 10, 15, 20, 25)) +
  labs(title=NULL,x= "", y = "Latency (min)") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 30, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 30, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 25, face = "bold"),
                                 legend.position = "none")
latency_vertical_separate

### pooled probes

## horizontal arena
latency_horizontal_pooled <- behaviour_data %>%
  drop_na(latency_min) %>%
  filter(arena == "horizontal" & light_environment == "normal") %>%
  ggplot(aes(y= latency_min, x = group, fill = group)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#CC79A7")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi")))) +
  scale_y_continuous(limits = c(0, 26), breaks = c(0, 5, 10, 15, 20, 25)) +
  labs(title=NULL,x= "", y = "Latency (min)") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  #geom_signif(comparisons=list(c("control", "lens3")), annotations="ns", # adds a significance bar
              #map_signif_level = TRUE, textsize = 5, vjust = -0.1) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, vjust = -0.15),
        axis.title.y = element_text(size = 38, vjust = 1.8),
        axis.text.y = element_text(size = 30),
        axis.text.x = element_text(size = 30),
        legend.position = "none")
latency_horizontal_pooled

## vertical arena
latency_vertical_pooled <- behaviour_data %>%
  drop_na(latency_min) %>%
  filter(arena == "vertical" & light_environment == "normal") %>%
  ggplot(aes(y= latency_min, x = group, fill = group)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#CC79A7")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi")))) +
  scale_y_continuous(limits = c(0, 26), breaks = c(0, 5, 10, 15, 20, 25)) +
  labs(title=NULL,x= "", y = "Latency (min)") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  #geom_signif(comparisons=list(c("control", "lens3")), annotations="ns", 
              #map_signif_level = TRUE, textsize = 5, vjust = -0.1) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, vjust = -0.15),
        axis.title.y = element_text(size = 38, vjust = 1.8),
        axis.text.y = element_text(size = 30),
        axis.text.x = element_text(size = 30),
        legend.position = "none")
latency_vertical_pooled

```

## Dark conditions
96% LEDs covered and UV supplementation removed

### Missed strikes
Dark

```{r dark missed strikes}

## horizontal arena
dark_missed_strikes_horizontal <- behaviour_data %>%
  filter(arena == "horizontal" & light_environment == "dark") %>%
  ggplot(aes(y= missed_strikes, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "Missed Strikes") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20, face = "bold"),
                                 legend.position = "none")
dark_missed_strikes_horizontal

## vertical arena
dark_missed_strikes_vertical <- behaviour_data %>%
  filter(arena == "vertical" & light_environment == "dark") %>%
  ggplot(aes(y= missed_strikes, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "Missed Strikes") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20, face = "bold"),
                                 legend.position = "none")
dark_missed_strikes_vertical
```

### Succesful strikes
Dark

```{r dark succesful strikes}

## horizontal arena
dark_succesful_strikes_horizontal <- behaviour_data %>%
  filter(arena == "horizontal" & light_environment == "dark") %>%
  ggplot(aes(y= succesful_strikes, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "Succesful Strikes") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20, face = "bold"),
                                 legend.position = "none")
dark_succesful_strikes_horizontal

## vertical arena
dark_succesful_strikes_vertical <- behaviour_data %>%
  filter(arena == "vertical" & light_environment == "dark") %>%
  ggplot(aes(y= succesful_strikes, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "Succesful Strikes") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20, face = "bold"),
                                 legend.position = "none")
dark_succesful_strikes_vertical
```

### Hunting success
Dark

```{r dark hunting success}

### separate probes

## horizontal arena
dark_hunting_horizontal <- behaviour_data %>%
  filter(arena == "horizontal" & light_environment == "dark") %>%
  ggplot(aes(y = hunting_success, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA, position = position_dodge(0.8)) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "% Hunting success") +
  geom_dotplot(aes(fill = NULL, alpha = 0.1), binaxis = 'y', stackdir = 'center',
        stackratio = 1.5, dotsize = 1.4, position = position_dodge(0.8)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 40, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 30, face = "bold"),
        legend.position = "none")
dark_hunting_horizontal

## vertical
dark_hunting_vertical <- behaviour_data %>%
  filter(arena == "vertical" & light_environment == "dark") %>%
  ggplot(aes(y = hunting_success, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA, position = position_dodge(0.8)) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "% Hunting success") +
  geom_dotplot(aes(fill = NULL, alpha = 0.1), binaxis = 'y', stackdir = 'center',
        stackratio = 1.5, dotsize = 1.4, position = position_dodge(0.8)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 40, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 30, face = "bold"),
        legend.position = "none")
dark_hunting_vertical

### pooled probes

## horizontal arena
dark_hunting_horizontal_pooled <- behaviour_data %>%
  filter(arena == "horizontal" & light_environment == "dark") %>%
  ggplot(aes(y = hunting_success, x = group, fill = group)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA, position = position_dodge(0.8)) +
  scale_fill_manual(values=c("#009E73", "#CC79A7")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi")))) +
  scale_y_continuous(limits = c(0, 100), 
                     breaks = c(0, 25, 50, 75, 100)) +
  labs(title=NULL,x= "", y = "% Hunting success") +
  geom_dotplot(aes(fill = NULL, alpha = 0.1), binaxis = 'y', stackdir = 'center',
        stackratio = 1.5, dotsize = 1.5, position = position_dodge(0.8)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, vjust = -0.15),
        axis.title.y = element_text(size = 38, vjust = 1.8),
        axis.text.y = element_text(size = 30),
        axis.text.x = element_text(size = 30),
        legend.position = "none")
dark_hunting_horizontal_pooled

## vertical
dark_hunting_vertical_pooled <- behaviour_data %>%
  filter(arena == "vertical" & light_environment == "dark") %>%
  ggplot(aes(y = hunting_success, x = group, fill = group)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA, position = position_dodge(0.8)) +
  scale_fill_manual(values=c("#009E73", "#CC79A7")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi")))) +
  scale_y_continuous(limits = c(0, 100),
                     breaks = c(0, 25, 50, 75, 100)) +
  labs(title=NULL,x= "", y = "% Hunting success") +
  geom_dotplot(aes(fill = NULL, alpha = 0.1), binaxis = 'y', stackdir = 'center',
        stackratio = 1.5, dotsize = 1.4, position = position_dodge(0.8)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, vjust = -0.15),
        axis.title.y = element_text(size = 38, vjust = 1.8),
        axis.text.y = element_text(size = 30),
        axis.text.x = element_text(size = 30),
        legend.position = "none")
dark_hunting_vertical_pooled

```

### Total strikes
Dark

```{r dark total strikes}

# horizontal arena
dark_total_strikes_horizontal <- behaviour_data %>%
  filter(arena == "horizontal" & light_environment == "dark") %>%
  ggplot(aes(y= total_strikes, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "Total Strikes") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20, face = "bold"),
                                 legend.position = "none")
dark_total_strikes_horizontal

# vertical arena
dark_total_strikes_vertical <- behaviour_data %>%
  filter(arena == "vertical" & light_environment == "dark") %>%
  ggplot(aes(y= total_strikes, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  labs(title=NULL,x= "", y = "Total Strikes") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 25, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 25, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 20, face = "bold"),
                                 legend.position = "none")
dark_total_strikes_vertical
```

### Latency
Dark

```{r dark latency}

### separate probes

## horizontal arena
dark_latency_horizontal_separate <- behaviour_data %>%
  drop_na(latency_min) %>%
  filter(arena == "horizontal" & light_environment == "dark") %>%
  ggplot(aes(y= latency_min, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  scale_y_continuous(limits = c(0, 26), breaks = c(0, 5, 10, 15, 20, 25)) +
  labs(title=NULL,x= "", y = "Latency (min)") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 40, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 30, face = "bold"),
                                 legend.position = "none")
dark_latency_horizontal_separate

## vertical arena
dark_latency_vertical_separate <- behaviour_data %>%
  drop_na(latency_min) %>%
  filter(arena == "vertical" & light_environment == "dark") %>%
  ggplot(aes(y= latency_min, x = probe, fill = probe)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#E69F00", "#D55E00")) +
  scale_x_discrete(labels = c('Control', 'lens3RNAi P1', 'lens3RNAi P2')) +
  scale_y_continuous(limits = c(0, 26), breaks = c(0, 5, 10, 15, 20, 25)) +
  labs(title=NULL,x= "", y = "Latency (min)") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  theme_classic() +
  theme(axis.title.x = element_text(size = 30, face = "bold", vjust = -0.15),
        axis.title.y = element_text(size = 30, face = "bold", vjust = 1.8),
        axis.text = element_text(size = 25, face = "bold"),
                                 legend.position = "none")
dark_latency_vertical_separate

### pooled probes

## horizontal arena
dark_latency_horizontal_pooled <- behaviour_data %>%
  drop_na(latency_min) %>%
  filter(arena == "horizontal" & light_environment == "dark") %>%
  ggplot(aes(y= latency_min, x = group, fill = group)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#CC79A7")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi")))) +
  scale_y_continuous(limits = c(0, 27), breaks = c(0, 5, 10, 15, 20, 25)) +
  labs(title=NULL,x= "", y = "Latency (min)") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  geom_signif(comparisons=list(c("control", "lens3")), annotations="*", # adds a significance star
              map_signif_level = TRUE, textsize = 10, 
              vjust = 0.5,
              margin_top = 0.04) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, vjust = -0.15),
        axis.title.y = element_text(size = 38, vjust = 1.8),
        axis.text = element_text(size = 30),
                                 legend.position = "none")
dark_latency_horizontal_pooled

## vertical arena
dark_latency_vertical_pooled <- behaviour_data %>%
  drop_na(latency_min) %>%
  filter(arena == "vertical" & light_environment == "dark") %>%
  ggplot(aes(y= latency_min, x = group, fill = group)) +
  geom_boxplot(alpha = 0.8, outlier.shape = NA) +
  scale_fill_manual(values=c("#009E73", "#CC79A7")) +
  scale_x_discrete(labels = c(expression("Control", 
                                         paste(italic("lens3"), "RNAi")))) +
  scale_y_continuous(limits = c(0, 27), breaks = c(0, 5, 10, 15, 20, 25)) +
  labs(title=NULL,x= "", y = "Latency (min)") +
  geom_jitter(shape = 19, size = 3.5, alpha = 0.7, position = position_jitter(0.1)) + 
  geom_signif(comparisons=list(c("control", "lens3")), annotations="*", # adds a significance star
              map_signif_level = TRUE, textsize = 10, 
              vjust = 0.5,
              margin_top = 0.35) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 40, vjust = -0.15),
        axis.title.y = element_text(size = 38, vjust = 1.8),
        axis.text = element_text(size = 30),
                                 legend.position = "none")
dark_latency_vertical_pooled
```

# Figure 7
```{r figure 7: behaviour}

#make latency panels 
fig.a <- latency_horizontal_pooled +
  labs(y = NULL) +
  theme(legend.position = "none",
        plot.margin = unit(c(1.5, 0, 0, 1), "cm")) + #The order of edges for plot.margin is unit(c(top, right, bottom, left), units) 
  coord_cartesian(clip = "off") 

fig.a

fig.b <- latency_vertical_pooled +
  labs(y = NULL) +
  theme(legend.position = "none",
        plot.margin = unit(c(1.5, 0, 0, 0.5), "cm")) +
  coord_cartesian(clip = "off")

fig.b

fig.c <- dark_latency_horizontal_pooled +
  labs(y = NULL) +
  theme(legend.position = "none",
        plot.margin = unit(c(1.5, 0, 0, 1), "cm")) +
  coord_cartesian(clip = "off") 

fig.c

fig.d <- dark_latency_vertical_pooled +
  labs(y = NULL) +
  theme(legend.position = "none",
        plot.margin = unit(c(1.5, 0, 0, 0.5), "cm")) +
  coord_cartesian(clip = "off")

fig.d

# use cowplot package to nicely plot the graphs together 
latency_plot <- plot_grid(fig.a, fig.c, fig.b, fig.d, 
                   #list of plots to arrange in grid
                   rel_widths = c(6, 6),
                   axis = c("b", "b", "b", "b"),
                   ncol = 2,
                   align = "v",
                   labels = c("A","B","C","D"), # panel labels for figure
                   label_size = 23,
                   hjust = -0.45, # adjustment for panel labels
                   vjust = 1.4)

plot_grid(latency_plot)

y.grob.latency <- textGrob("Latency (min)",
                   just = "left", hjust = 0.4, vjust = 0.4,
                   gp = gpar(fontface = "bold",
                             fontsize = 30), rot = 90)

latency_plot_axis <- grid.arrange(arrangeGrob(latency_plot, left = y.grob.latency)) #adding common y axis


# make hunting success panels 
fig.e <- hunting_horizontal_pooled +
  labs(y = NULL) +
  theme(legend.position = "none",
        plot.margin = unit(c(1.5, 0, 0, 1), "cm")) + #The order of edges for plot.margin is unit(c(top, right, bottom, left), units) 
  coord_cartesian(clip = "off") 

fig.e

fig.f <- hunting_vertical_pooled +
  labs(y = NULL) +
  theme(legend.position = "none",
        plot.margin = unit(c(1.5, 0, 0, 0.5), "cm")) +
  coord_cartesian(clip = "off")

fig.f

fig.g <- dark_hunting_horizontal_pooled +
  labs(y = NULL) +
  theme(legend.position = "none",
        plot.margin = unit(c(1.5, 0, 0, 1), "cm")) +
  coord_cartesian(clip = "off") 

fig.g

fig.h <- dark_hunting_vertical_pooled +
  labs(y = NULL) +
  theme(legend.position = "none",
        plot.margin = unit(c(1.5, 0, 0, 0.5), "cm")) +
  coord_cartesian(clip = "off")

fig.h

# use cowplot package to nicely plot the graphs together
hunting_plot <- plot_grid(fig.e, fig.g, fig.f, fig.h, 
                   # list of plots to arrange in grid
                   rel_widths = c(6, 6),
                   axis = c("b", "b", "b", "b"),
                   ncol = 2,
                   align = "v",
                   labels = c("E","F","G","H"), # panel labels for figure
                   label_size = 23,
                   hjust = -0.45, # adjustment for panel labels
                   vjust = 1.4) 

plot_grid(hunting_plot)

y.grob.hunting <- textGrob("% Hunting success",
                   just = "left", hjust = 0.4, vjust = 0.4,
                   gp = gpar(fontface = "bold",
                             fontsize = 30), rot = 90)

hunting_plot_axis <- grid.arrange(arrangeGrob(hunting_plot, left = y.grob.hunting)) # adding common y axis

behaviour_plots <- grid.arrange(arrangeGrob(latency_plot_axis, hunting_plot_axis, 
                                            nrow = 2,
                                            heights = c(6.5, 6)))

plot_grid(behaviour_plots)

# Adding larval silhouette and light icons

## making space for icons
### adding a column for larva silhouettes
behaviour_figure <- plot_grid(NULL, behaviour_plots, NULL,
                 rel_widths = c(0.2, 6, 1.2),
                 ncol = 3)

# adding a row for the light arena icons
behaviour_figure <- plot_grid(NULL, behaviour_figure,
                  rel_heights = c(0.9, 6),
                  nrow = 2)

plot_grid(behaviour_figure)

# loading .png icons into environment
larva_h_icon <- "larva_horizontal.png"
larva_v_icon <- "larva_vertical.png"
bright_icon <- "bright_light.png"
dim_icon <- "dim_light.png"
uv_icon <- "uv_light.png"

bright_label <- expression("Absolute spectral irradiance (µW/cm"^2*"/nm)")

# drawing icons onto the final figure
behaviour_final <- ggdraw() +
  draw_plot(behaviour_figure) +
  # horizontal larva silhouettes
  draw_image(image = larva_h_icon, x = 0.41, y =  0.28, scale = 0.13) +
  draw_image(image = larva_h_icon, x = 0.41, y =  -0.17, scale = 0.13) +
  # vertical larva silhouettes
  draw_image(image = larva_v_icon, x = 0.41, y =  0.045, scale = 0.13) +
  draw_image(image = larva_v_icon, x = 0.41, y =  -0.39, scale = 0.13) +
  # bright, dim & uv light vectors
  draw_image(image = bright_icon, x = -0.27, y =  0.465, scale = 0.083) +
  draw_image(image = dim_icon, x = 0.165, y =  0.457, scale = 0.075) +
  draw_image(image = uv_icon, x = -0.18, y = 0.466, scale = 0.083) +
  # labels for photon count
  draw_label(label = expression(bold("6.27 - 7.03 x 10"^"14"*" photons/cm"^"2"*"/s")),
             x = 0.27, y = 0.9,
             size = 20) +
  draw_label(label = expression(bold("9.29 x 10"^"12"*" photons/cm"^"2"*"/s")),
             x = 0.67, y = 0.9,
             size = 20)

behaviour_final

# print graph
pdf(file = "Figures/figure_7.pdf", width = 15,  height = 18)
behaviour_final
dev.off()

```

## Spectrophotometer

```{r spec plots}

# Defining nm_to_RGB. 
# This is a function which converts a wavelength of light into a hex-string to represent a colour
nm_to_RGB <- function(wavelengths){
  sapply(wavelengths, function(wavelength) {
  red <- green <- blue <- 0  
  if((wavelength >= 380) & (wavelength < 440)){
    red <- -(wavelength - 440) / (440 - 380)
    blue <- 1
  }else if((wavelength >= 440) & (wavelength<490)){
    green <- (wavelength - 440) / (490 - 440)
    blue <- 1
  }else if((wavelength >= 490) && (wavelength<510)){
    green <- 1
    blue = -(wavelength - 510) / (510 - 490)
  }else if((wavelength >= 510) && (wavelength<580)){
    red = (wavelength - 510) / (580 - 510)
    green <- 1
  }else if((wavelength >= 580) && (wavelength<645)){
    red = 1
    green <- -(wavelength - 645) / (645 - 580)
  }else if((wavelength >= 645) && (wavelength<781)){
    red = 1
  }
  if((wavelength >= 380) && (wavelength<420)){
    fac <- 0.3 + 0.7*(wavelength - 380) / (420 - 380)
  }else if((wavelength >= 420) && (wavelength<701)){
    fac <- 1
  }else if((wavelength >= 701) && (wavelength<781)){
    fac <- 0.3 + 0.7*(780 - wavelength) / (780 - 700)
  }else{
    fac <- 0
  }
  do.call(rgb, as.list((c(red, green, blue) * fac)^0.8))
  })
}

spec_data <- spec_data %>%
  mutate(across(c("absolute_irradiance"), abs))

## bright
# horizontal 
horizontal_bright_spec_colour_unsmooth <- spec_data %>%
  filter(arena == "horizontal" & light_environment == "bright") %>%
  ggplot(aes(x = wavelength, y = absolute_irradiance)) +
  geom_segment(aes(xend = wavelength, yend = 0, 
                   colour = nm_to_RGB(wavelength))) +
  geom_line() +
  scale_colour_identity() +
  xlab("Wavelength (nm)") + ylab(label= expression("Absolute spectral irradiance (µW/cm"^2*"/nm)")) +
  scale_x_continuous(limits = c(300, 750), 
                     breaks = c(350, 400, 450, 500, 550, 600, 650, 700)) +
  scale_y_continuous(limits = c(0, 7.5),
                     expand = c(0, 0), # removes space between plotted data and y axis
                     breaks = c(1, 2, 3, 4, 5, 6)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 20, vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")

horizontal_bright_spec_colour_unsmooth

# The only issue here is that the colour gradient is plotted using geom_segment which uses exact values. This is not an issue with geom_line which also plots exact values. But the problem arises with the dark spectra where the resolution of the spec is so low that the data looks choppy and disgusting with geom_line. Here a smoothed line is needed which geom_spline is perfect for. However, geom_spline then doesn't match with geom_segment and makes a giant mess. A fix, albeit quite hacky and unelegant, is to extract the fit values from the spline and plot them separately so the geom_segment matches:

# horizontal
# no gradient
horizontal_bright_spec <- spec_data %>%
  filter(arena == "horizontal" & light_environment == "bright") %>%
  ggplot(aes(x = wavelength, y = absolute_irradiance)) +
  geom_spline() +
  xlab("Wavelength (nm)") + ylab(label= expression("Absolute spectral irradiance (µW/cm"^2*"/nm)")) +
  scale_x_continuous(limits = c(300, 750), 
                     breaks = c(350, 400, 450, 500, 550, 600, 650, 700)) +
  scale_y_continuous(limits = c(0, 7.5),
                     expand = c(0, 0), # removes space between plotted data and y axis
                     breaks = c(1, 2, 3, 4, 5, 6, 7)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 20, vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")

horizontal_bright_spec

# gradient
hor_bright_spec_df<- ggplot_build(horizontal_bright_spec)$data[[1]] # extracting the spline values from the plot into a new dataframe

horizontal_bright_spec_colour <- hor_bright_spec_df %>% # plotting extracted values
  ggplot(aes(x = x, y = y)) +
  geom_segment(aes(xend = x, yend = 0, 
                   colour = nm_to_RGB(x))) +
  geom_line() +
  scale_colour_identity() +
  xlab("Wavelength (nm)") + ylab(label= expression("Absolute spectral irradiance (µW/cm"^2*"/nm)")) +
  scale_x_continuous(limits = c(300, 750), 
                     breaks = c(350, 400, 450, 500, 550, 600, 650, 700)) +
  scale_y_continuous(limits = c(0, 7.5),
                     expand = c(0, 0), # removes space between plotted data and y axis
                     breaks = c(1, 2, 3, 4, 5, 6, 7)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 20, vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")

horizontal_bright_spec_colour

# vertical
# no gradient
vertical_bright_spec <- spec_data %>%
  filter(arena == "vertical" & light_environment == "bright") %>%
  ggplot(aes(x = wavelength, y = absolute_irradiance)) +
  geom_spline() +
  xlab("Wavelength (nm)") + ylab(label= expression("Absolute spectral irradiance (µW/cm"^2*"/nm)")) +
  scale_x_continuous(limits = c(300, 750), 
                     breaks = c(350, 400, 450, 500, 550, 600, 650, 700)) +
  scale_y_continuous(limits = c(0, 7.5),
                     expand = c(0, 0), # removes space between plotted data and y axis
                     breaks = c(1, 2, 3, 4, 5, 6, 7)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 20, vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")

vertical_bright_spec

# gradient
ver_bright_spec_df<- ggplot_build(vertical_bright_spec)$data[[1]] # extracting the spline values from the plot into a new dataframe

vertical_bright_spec_colour <- ver_bright_spec_df %>% # plotting extracted values
  ggplot(aes(x = x, y = y)) +
  geom_segment(aes(xend = x, yend = 0, 
                   colour = nm_to_RGB(x))) +
  geom_line() +
  scale_colour_identity() +
  xlab("Wavelength (nm)") + ylab(label= expression("Absolute spectral irradiance (µW/cm"^2*"/nm)")) +
  scale_x_continuous(limits = c(300, 750), 
                     breaks = c(350, 400, 450, 500, 550, 600, 650, 700)) +
  scale_y_continuous(limits = c(0, 7.5),
                     expand = c(0, 0), # removes space between plotted data and y axis
                     breaks = c(1, 2, 3, 4, 5, 6, 7)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 20, vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")

vertical_bright_spec_colour

## dark
# horizontal 
# no gradient
horizontal_dark_spec <- spec_data %>%
  filter(arena == "horizontal" & light_environment == "dark") %>%
  ggplot(aes(x = wavelength, y = absolute_irradiance)) +
  geom_spline() +
  scale_x_continuous(limits = c(300, 750), 
                     breaks = c(350, 400, 450, 500, 550, 600, 650, 700)) +
  scale_y_continuous(limits = c(0, 0.065),
                     expand = c(0, 0)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 20, vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")
horizontal_dark_spec

# gradient
hor_dark_spec_df<- ggplot_build(horizontal_dark_spec)$data[[1]] # extracting the spline values from the plot into a new dataframe

horizontal_dark_spec_colour <- hor_dark_spec_df %>% # plotting extracted values
  ggplot(aes(x = x, y = y)) +
  geom_segment(aes(xend = x, yend = 0, 
                   colour = nm_to_RGB(x))) +
  geom_line() +
  scale_colour_identity() +
  xlab("Wavelength (nm)") + ylab(label= expression("Absolute spectral irradiance (µW/cm"^2*"/nm)")) +
  scale_x_continuous(limits = c(300, 750), 
                     breaks = c(350, 400, 450, 500, 550, 600, 650, 700)) +
  scale_y_continuous(limits = c(0, 0.065),
                     expand = c(0, 0)) + # removes space between plotted data and y axis
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 20, vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")

horizontal_dark_spec_colour

# vertical
# no gradient
vertical_dark_spec <- spec_data %>%
  filter(arena == "vertical" & light_environment == "dark") %>%
  ggplot(aes(x = wavelength, y = absolute_irradiance)) +
  geom_spline() +
  xlab("Wavelength (nm)") + ylab(label= expression("Absolute spectral irradiance (µW/cm"^2*"/nm)")) +
  scale_x_continuous(limits = c(300, 750), 
                     breaks = c(350, 400, 450, 500, 550, 600, 650, 700)) +
  scale_y_continuous(limits = c(0, 0.065),
                     expand = c(0, 0)) +
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 20, vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")
vertical_dark_spec

# gradient
ver_dark_spec_df<- ggplot_build(vertical_dark_spec)$data[[1]] # extracting the spline values from the plot into a new dataframe

vertical_dark_spec_colour <- ver_dark_spec_df %>% # plotting extracted values
  ggplot(aes(x = x, y = y)) +
  geom_segment(aes(xend = x, yend = 0, 
                   colour = nm_to_RGB(x))) +
  geom_line() +
  scale_colour_identity() +
  xlab("Wavelength (nm)") + ylab(label= expression("Absolute spectral irradiance (µW/cm"^2*"/nm)")) +
  scale_x_continuous(limits = c(300, 750), 
                     breaks = c(350, 400, 450, 500, 550, 600, 650, 700)) +
  scale_y_continuous(limits = c(0, 0.065),
                     expand = c(0, 0)) + # removes space between plotted data and y axis
  theme_classic() +
  theme(axis.title.x = element_text(size = 20, vjust = -0.15),
        axis.title.y = element_text(size = 20, vjust = 1.8),
        axis.text = element_text(size = 17),
                                 legend.position = "none")

vertical_dark_spec_colour
```

#Figure S3
```{r figure S3: spectra}

# Making panels
fig.a <- horizontal_bright_spec_colour +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank())

fig.b <- vertical_bright_spec_colour +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank())

fig.c <- horizontal_dark_spec_colour +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank())

fig.d <- vertical_dark_spec_colour +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_blank())

# Building the figure
spec_figure <- plot_grid(fig.a, fig.b, fig.c, fig.d,
           align = 'vh', # align horizontally and vertically
           axis = 'lb',
           labels = c("A","B","C","D"), # panel labels for figure
           label_size = 28,
           hjust = -.01, # adjustment for panel labels
           vjust = -0.03,
           nrow = 2) + # number of rows in grids
           theme(plot.margin = unit(c(40, 40, 25, 25), "points")) # The order of edges for plot.margin is unit(c(top, right, bottom, left), units)
  
spec_figure

# Creating common axis labels
y.grob <- textGrob(label= expression(bold("Absolute spectral irradiance (µW/cm"^"2"*"/nm)")), 
                   gp = gpar(fontsize = 25), rot = 90, 
                             vjust = 0.8,
                             hjust = 0.5)

x.grob <- textGrob("Wavelength (nm)",
                   gp = gpar(fontface = "bold",
                             fontsize = 25),
                             vjust = -0.3)

# adding common x axis label
spec_figure_xlab <- plot_grid(spec_figure, x.grob,
                                 ncol = 1, rel_heights = c(1, 0.03))
plot_grid(spec_figure_xlab)

# adding common y axis label
spec_figure_labs <- plot_grid(y.grob, spec_figure_xlab,
                                 ncol = 2, rel_widths = c(0.03, 1))

spec_figure_margins <- plot_grid(spec_figure_labs) +
                       theme(plot.margin = unit(c(20, 20, 0, 0), "points")) # The order of edges for plot.margin is unit(c(top, right, bottom, left), units)

# rotating the horizontal larva icon by 90°
larva_icon_magick <- image_read('larva_horizontal.png') # loading .png as a magick object for rotation
larva_h_icon_rotate <- image_rotate(larva_icon_magick, 270) 

# drawing icons onto the plot
spec_figure_final<- ggdraw() +
  draw_plot(spec_figure_margins) +
    # bright & dim light vectors
    draw_image(image = bright_icon, x = 0.47, y =  0.3, scale = 0.083) +
    draw_image(image = uv_icon, x = 0.47, y = 0.20, scale = 0.083) + 
    draw_image(image = dim_icon, x = 0.47, y =  -0.2, scale = 0.075) +
    # horizontal larva silhouettes
    draw_image(image = larva_h_icon_rotate, x = -0.19, y =  0.45, scale = 0.13) +
    draw_image(image = larva_v_icon, x = 0.25, y =  0.441, scale = 0.13)

spec_figure_final

# Print graph
cairo_pdf(file = "Figures/figure_S3.pdf", width = 18,  height = 12) # cairo_pdf is able to handle special characters\
spec_figure_final
dev.off()

```

# Statistics

## qPCR

```{r qPCR stats}

## Test for normality

# Building the linear model
qPCR_model <- lm(log_fold_gene_expression ~ probe, data = qpcr_data)

# Creating a QQ plot of residuals
ggqqplot(residuals(qPCR_model))

# Test for normality
shapiro_test(residuals(qPCR_model))
# p > 0.05 so assuming normality

# Using an ANOVA test
res_qPCR_aov <- anova_test(log_fold_gene_expression ~ probe, data = qpcr_data)
res_qPCR_aov

summary(res_qPCR_aov)
# Significant reduction in log fold lens3 expression for both probes compared to controls

```

## Edge sharpness series

```{r edge sharpness stats}

## Test for normality

# Building the linear model
edge_model <- lm(maximum_sharpness ~ probe, data = hang_data)

# Creating a QQ plot of residuals
ggqqplot(residuals(edge_model))

# Test for normality
shapiro_test(residuals(edge_model))
# p < 0.05 so not assuming normality

# Checking normality assumption by groups
hang_data %>%
  group_by(probe) %>%
  shapiro_test(maximum_sharpness)
# again p < 0.05 so not assuming normality

# Creating QQ plots for each group level
ggqqplot(hang_data, "maximum_sharpness", facet.by = "probe")
# Control and Probe 1 look normal. Probe 2 is kinda wonky. But will proceed with Gaussian stats

# Using a Kruskal Wallis test
res_edge_kw <- hang_data %>%
  group_by(eye) %>%
  kruskal_test(maximum_sharpness ~ probe)

res_edge_kw

# Significant changes in maximum edge sharpness with lens3 knockdown - all eyes

```

## Back focal length

```{r back focal length stats}

## Test for normality

# Building the linear model
focal_model <- lm(peak_2_um ~ probe, data = hang_data)

# Creating a QQ plot of residuals
ggqqplot(residuals(focal_model))

# Test for normality
shapiro_test(residuals(focal_model))
# p > 0.05 so assuming normality

# Checking normality assumption by groups
hang_data %>%
  group_by(probe) %>%
  shapiro_test(peak_2_um)
# again p > 0.05 so assuming normality

# Creating QQ plots for each group level
ggqqplot(hang_data, "peak_2_um", facet.by = "probe")
# Control and Probe 1 look normal. Probe 2 is kinda wonky. But will proceed with Gaussian stats

# Using an ANOVA test
res_focal_aov <- compare_means(peak_2_um ~ probe, data = hang_data, 
              group.by = "eye", method = 'anova')
res_focal_aov

res_focal_aov <- hang_data %>%
  group_by(eye) %>%
  anova_test(peak_2_um ~ probe)

res_focal_aov

summary(res_focal_aov)
# No significant changes in back focal length with lens3 knockdown - all eyes

```

## Ophthalmoscope

```{r ophthalmoscope stats}

## Test for normality

# Building the linear model
oph_model <- lm(bestimagereciprocal ~ probe, data = oph_data)

# Creating a QQ plot of residuals
ggqqplot(residuals(oph_model))

# Test for normality
shapiro_test(residuals(oph_model))
# p < 0.05 so can't assume normality

# Checking normality assumption by groups
oph_data %>%
  group_by(probe) %>%
  shapiro_test(bestimagereciprocal)
# again p < 0.05 so can't assume normality

# Creating QQ plots for each group level
ggqqplot(oph_data, "bestimagereciprocal", facet.by = "probe")
# Probe 2 (lens3_249) is the only one that looks normal. Other two are kinda wonky

# Using a Kruskal-Wallis test
res_oph_kw <- oph_data %>%
  group_by(eye) %>%
  kruskal_test(bestimagereciprocal ~ probe)

res_oph_kw

summary(res_oph_kw)
# No significant changes in focused state with lens3 knockdown - all eyes

```

## Behaviour

## Normal light conditions
### Latency pooled probes

```{r normal light latency}

## Horizontal
# Building the linear model
pooled_latency_model_hor_light <- lm(latency_min ~ group, 
                                    data = behaviour_data, 
                                    subset = (light_environment == 'normal' & arena == 'horizontal'))

# Creating a QQ plot of residuals
ggqqplot(residuals(pooled_latency_model_hor_light))

# Test for normality
shapiro_test(residuals(pooled_latency_model_hor_light))
# p < 0.05 so can't assume normality

# Checking normality assumption by groups
behaviour_data %>%
  filter(light_environment == 'normal' & arena == 'horizontal') %>%
  group_by(group) %>%
  shapiro_test(latency_min)
# both groups not normal

## Vertical
# Building the linear model
pooled_latency_model_ver_light <- lm(latency_min ~ group, 
                                    data = behaviour_data, 
                                    subset = (light_environment == 'normal' & arena == 'vertical'))

# Creating a QQ plot of residuals
ggqqplot(residuals(pooled_latency_model_ver_light))

# Test for normality
shapiro_test(residuals(pooled_latency_model_ver_light))
# p > 0.05 so can't assume normality

# Checking normality assumption by groups
behaviour_data %>%
  filter(light_environment == 'normal' & arena == 'vertical') %>%
  group_by(group) %>%
  shapiro_test(latency_min)
# both groups not normal

## non-parametric
# horizontal test
res_latency_light_horizontal_pooled_wil <- behaviour_data %>%
  filter(light_environment == 'normal', arena == 'horizontal') %>%
  wilcox_test(latency_min ~ group)

res_latency_light_horizontal_pooled_wil
# not significant

# vertical test
res_latency_light_vertical_pooled_wil <- behaviour_data %>%
  filter(light_environment == 'normal', arena == 'vertical') %>%
  wilcox_test(latency_min ~ group)

res_latency_light_vertical_pooled_wil
# not significant

```

### Hunting success pooled probes

```{r normal light hunting success}

## Horizontal
# Building the linear model
pooled_hunting_model_hor_light <- lm(hunting_success ~ group, 
                                    data = behaviour_data, 
                                    subset = (light_environment == 'normal' & arena == 'horizontal'))

# Creating a QQ plot of residuals
ggqqplot(residuals(pooled_hunting_model_hor_light))

# Test for normality
shapiro_test(residuals(pooled_hunting_model_hor_light))
# p < 0.05 so can't assume normality - as expected 

## Vertical
# Building the linear model
pooled_hunting_model_ver_light <- lm(hunting_success ~ group, 
                                    data = behaviour_data, 
                                    subset = (light_environment == 'normal' & arena == 'vertical'))

# Creating a QQ plot of residuals
ggqqplot(residuals(pooled_hunting_model_ver_light))

# Test for normality
shapiro_test(residuals(pooled_hunting_model_ver_light))
# p > 0.05 so can't assume normality - as expected 

# horizontal test
res_hunting_light_horizontal_pooled_wil <- behaviour_data %>%
  filter(light_environment == 'normal', arena == 'horizontal') %>%
  wilcox_test(hunting_success ~ group)

res_hunting_light_horizontal_pooled_wil
# not significant

# vertical test
res_hunting_light_vertical_pooled_wil <- behaviour_data %>%
  filter(light_environment == 'normal', arena == 'vertical') %>%
  wilcox_test(hunting_success ~ group)

res_hunting_light_vertical_pooled_wil
# not significant
```

## Dark conditions
### Latency pooled probes

```{r dark latency stats}

## Horizontal
# Building the linear model
pooled_latency_model_hor_dark <- lm(latency_min ~ group, 
                                    data = behaviour_data, 
                                    subset = (light_environment == 'dark' & arena == 'horizontal'))

# Creating a QQ plot of residuals
ggqqplot(residuals(pooled_latency_model_hor_dark))

# Test for normality
shapiro_test(residuals(pooled_latency_model_hor_dark))
# p < 0.05 so can't assume normality

# Checking normality assumption by groups
behaviour_data %>%
  filter(light_environment == 'dark' & arena == 'horizontal') %>%
  group_by(group) %>%
  shapiro_test(latency_min)
# controls normal but lens3 not

## Vertical
# Building the linear model
pooled_latency_model_ver_dark <- lm(latency_min ~ group, 
                                    data = behaviour_data, 
                                    subset = (light_environment == 'dark' & arena == 'vertical'))

# Creating a QQ plot of residuals
ggqqplot(residuals(pooled_latency_model_ver_dark))

# Test for normality
shapiro_test(residuals(pooled_latency_model_ver_dark))
# p > 0.05 so *can* assume normality

# Checking normality assumption by groups
behaviour_data %>%
  filter(light_environment == 'dark' & arena == 'vertical') %>%
  group_by(group) %>%
  shapiro_test(latency_min)
# not normal

## non-parametric
# horizontal test
res_latency_dark_horizontal_pooled_wil <- behaviour_data %>%
  filter(light_environment == 'dark', arena == 'horizontal') %>%
  wilcox_test(latency_min ~ group)

res_latency_dark_horizontal_pooled_wil
# significant!!

# vertical test
res_latency_dark_vertical_pooled_wil <- behaviour_data %>%
  filter(light_environment == 'dark', arena == 'vertical') %>%
  wilcox_test(latency_min ~ group)

res_latency_dark_vertical_pooled_wil
# significant!!

```

### Hunting success pooled probes
```{r dark hunting success}

## Horizontal
# Building the linear model
pooled_hunting_model_hor_dark <- lm(hunting_success ~ group, 
                                    data = behaviour_data, 
                                    subset = (light_environment == 'dark' & arena == 'horizontal'))

# Creating a QQ plot of residuals
ggqqplot(residuals(pooled_hunting_model_hor_dark))

# Test for normality
shapiro_test(residuals(pooled_hunting_model_hor_dark))
# p < 0.05 so can't assume normality - as expected 


## Vertical
# Building the linear model
pooled_hunting_model_ver_dark <- lm(hunting_success ~ group, 
                                    data = behaviour_data, 
                                    subset = (light_environment == 'dark' & arena == 'vertical'))

# Creating a QQ plot of residuals
ggqqplot(residuals(pooled_hunting_model_ver_dark))

# Test for normality
shapiro_test(residuals(pooled_hunting_model_ver_dark))
# p > 0.05 so can't assume normality - as expected 

# horizontal test
res_hunting_dark_horizontal_pooled_wil <- behaviour_data %>%
  filter(light_environment == 'dark', arena == 'horizontal') %>%
  wilcox_test(hunting_success ~ group)

res_hunting_dark_horizontal_pooled_wil
# not significant

# vertical test
res_hunting_dark_vertical_pooled_wil <- behaviour_data %>%
  filter(light_environment == 'dark', arena == 'vertical') %>%
  wilcox_test(hunting_success ~ group)

res_hunting_dark_vertical_pooled_wil
# not significant

```